src/System.Management.Automation/engine/hostifaces/HostUtilities.cs

```diff
@@ -522,8 +522,18 @@ internal static string RemoveGuidFromMessage(string message, out bool matchPatte
         internal static string RemoveIdentifierInfoFromMessage(string message, out bool matchPattern)
         {
             matchPattern = false;
-            if (string.IsNullOrEmpty(message))
+            if (message is null ||
+                message.Length < 40 ||
+                message[37] != '[' ||
+                message[36] != ':' ||
+                message[23] != '-' ||
+                message[18] != '-' ||
+                message[13] != '-' ||
+                message[8] != '-')
+            {
+                // Fast return if the message does not start with 'GUID:[]:'.
                 return message;
+            }
 
             const string pattern = @"^([\d\w]{8}\-[\d\w]{4}\-[\d\w]{4}\-[\d\w]{4}\-[\d\w]{12}:\[.*\]:).*";
```

Comment:
```suggestion
            const string pattern = @"^[\d\w]{8}\-[\d\w]{4}\-[\d\w]{4}\-[\d\w]{4}\-[\d\w]{12}:\[.*\]:";
```

Remove the unnecessary capturing group and refactor:
```csharp
message = message.Substring(matchResult.Length);`
```

Reply:
Eh, I don't know that is real format of the messages - perhaps we need the capture group.

Reply:
@SteveL-MSFT I mean line 548 can become `message = message.Substring(matchResult.Length);`